src/serialize.js

```diff
@@ -24,10 +23,10 @@ export default (tree, { search } = {}) =>
         },
         x => {
           if (x.children) {
-            x.children = _.flow(
-              _.reject('markedForDeletion'),
-              _.reject(x => x.children && !x.children.length)
-            )(x.children)
+            x.children = _.reject(
+              x => x.children && !x.children.length,
+              x.children
```

Comment:
While we're at it, I think we should just eliminate *all* node deletion by the client - leaving in the code that wipes empty groups still makes the returned tree not necessarily match up in index order